### PR TITLE
refactor: add update event processor stubs - WPB-8742

### PR DESCRIPTION
**Pull request by johnxnguyen**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-8742" title="WPB-8742" target="_blank">TaskWPB-8742</a>  Create UpdateEventRepository
  </td></table>
  <br />
 

### Issue

This PR adds stubs for event processing:
- One root update event processor (`UpdateEventProcessor`) receives one event at a time
- It switches over the event, forwards it to a more specialized event processor, e.g `ConversationEventProcessor`.
- The specialized processor may switch over the event, and forward it to even still more specialized processors.
- All event processing throws and is async

### Testing

Nothing to test yet.

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

### CODE REVIEW

**Review by agisilaos — APPROVED** [review]

File: WireDomain/Sources/WireDomain/Event Processing/FederationEventProcessor.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    /// - Parameter event: A federation update event.
+
+    func processEvent(_ event: FederationEvent) async throws
+
+}
+
+struct FederationEventProcessor {
+
+    func processEvent(_ event: FederationEvent) async throws {
+        assertionFailure("not implemented")
```
Comment (caldrian) [inline WireDomain/Sources/WireDomain/Event Processing/FederationEventProcessor.swift]: I usually add a `"yet"` if something is about to be added in the future.
"not implemented" to me sounds like something is not needed/relevant.
Maybe I'm a bit too sensitive.

**Review by caldrian — APPROVED** [review]
👍

File: WireDomain/Sources/WireDomain/Event Processing/ConversationEventProcessor.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    /// - Parameter event: A conversation update event.
+
+    func processEvent(_ event: ConversationEvent) async throws
+
+}
+
+struct ConversationEventProcessor {
+
+    func processEvent(_ event: ConversationEvent) async throws {
+        assertionFailure("not implemented")
```
Comment (johnxnguyen) [inline WireDomain/Sources/WireDomain/Event Processing/ConversationEventProcessor.swift]:
```suggestion
        assertionFailure("not implemented yet")
```

File: WireDomain/Sources/WireDomain/Event Processing/FeatureConfigEventProcessor.swift [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    /// - Parameter event: A feature config update event.
+
+    func processEvent(_ event: FeatureConfigEvent) async throws
+
+}
+
+struct FeatureConfigEventProcessor {
+
+    func processEvent(_ event: FeatureConfigEvent) async throws {
+        assertionFailure("not implemented")
```
Comment (johnxnguyen) [inline WireDomain/Sources/WireDomain/Event Processing/FeatureConfigEventProcessor.swift]:
```suggestion
        assertionFailure("not implemented yet")
```

File: WireDomain/Sources/WireDomain/Event Processing/FederationEventProcessor.swift (hunk 1 above)
Comment (johnxnguyen) [inline WireDomain/Sources/WireDomain/Event Processing/FederationEventProcessor.swift]:
```suggestion
        assertionFailure("not implemented yet")
```

File: WireDomain/Sources/WireDomain/Event Processing/TeamEventProcessor.swift [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    /// - Parameter event: A team update event.
+
+    func processEvent(_ event: TeamEvent) async throws
+
+}
+
+struct TeamEventProcessor {
+
+    func processEvent(_ event: TeamEvent) async throws {
+        assertionFailure("not implemented")
```
Comment (johnxnguyen) [inline WireDomain/Sources/WireDomain/Event Processing/TeamEventProcessor.swift]:
```suggestion
        assertionFailure("not implemented yet")
```

File: WireDomain/Sources/WireDomain/Event Processing/UserEventProcessor.swift [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    /// - Parameter event: A user update event.
+
+    func processEvent(_ event: UserEvent) async throws
+
+}
+
+struct UserEventProcessor {
+
+    func processEvent(_ event: UserEvent) async throws {
+        assertionFailure("not implemented")
```
Comment (johnxnguyen) [inline WireDomain/Sources/WireDomain/Event Processing/UserEventProcessor.swift]:
```suggestion
        assertionFailure("not implemented yet")
```

File: WireDomain/Sources/WireDomain/Event Processing/FederationEventProcessor.swift (hunk 1 above)
Comment (johnxnguyen) [reply WireDomain/Sources/WireDomain/Event Processing/FederationEventProcessor.swift]: done

**Pull request merged**